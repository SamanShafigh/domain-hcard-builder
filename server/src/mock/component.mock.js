module.exports = {
  default: 'h1'
}