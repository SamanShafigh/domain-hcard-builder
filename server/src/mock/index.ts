import HCardT from './type';

export const mockEnv = <HCardT.Env>{
  STATIC_ROOT: 'root',
  PORT: 1234,
  RENDER_MODE: 'ssr',
  DB_URL: 'mongodb://localhost',
  DB_NAME: 'test'
};

export const mockConfig = {
  staticRoot: './client/dist',
  ssrPath: './dist/main.js',
  spaPath: './dist/_index.html',
  serverPort: 3030,
  renderMode: 'ssr',
  dbUri: 'mongodb://localhost',
  dbName: 'hcard',
}

export const mockUser = {
  givenName: 'givenName',
  surname: 'surname',
  email: 'email',
  phone: 'phone',
  houseNumber: 'houseNumber',
  street: 'street',
  suburb: 'suburb',
  state: 'state',
  postcode: 'postcode',
  country: 'country'
}

export const mockInvalidUser = {
  givenName: 'givenName',
  email: 'email',
  houseNumber: 'houseNumber',
  suburb: 'suburb',
  postcode: 'postcode',
  invalidData: 'some Invalid data'
}

export const mockCtx = {
  meta: {
    userId: '1234',
    noscript: true,
  }
}

export const mockCtxWithBody = {
  meta: {
    userId: '1234',
    noscript: true,
  },
  request: {
    body: mockUser
  }
}

export const mockDbDriver = {
  findOne: (model, query, options) => (Promise.resolve({ data: mockUser })),
  save: (model, query, data, options) => ({ model, query, data, options }),
  update: (model, query, data, options) => ({ model, query, data, options }),
}

export const mockMongoDbDriver = {
  collection: (model) => ({
    findOne: (query, options) => Promise.resolve({ model, query, options }),
    update: (query, data, options) => Promise.resolve({ model, query, data, options}),
  })
}

export const mockReactComponentPath = '../mock/component.mock.js'
export const mockHtmlPagePath = '../mock/page.mock.html'