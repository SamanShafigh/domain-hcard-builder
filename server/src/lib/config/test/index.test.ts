import { makeConfig } from '../index';
import { mockEnv } from '../../mock';

describe('Test config maker', () => {
  test('test config object with all provided env variables', () => {
    expect(makeConfig(mockEnv)).toMatchSnapshot();
  });

  test('test config object with only env.STATIC_ROOT variable', () => {
    const env = <HCardT.Env>{
      STATIC_ROOT: 'root',
    };
    expect(makeConfig(env)).toMatchSnapshot();
  });

  test('tes config object without any env variable', () => {
    const env = <HCardT.Env>{};
    expect(makeConfig(env)).toMatchSnapshot();
  });
});