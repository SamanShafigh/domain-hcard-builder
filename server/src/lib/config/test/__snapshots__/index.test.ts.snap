// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test config maker tes config object without any env variable 1`] = `
Object {
  "dbName": "hcard",
  "dbUri": "mongodb://192.168.99.101:27017/domain",
  "renderMode": "ssr",
  "serverPort": 3030,
  "spaPath": "../../../client/dist/_index.html",
  "ssrPath": "../../../client/dist/main.js",
  "staticRoot": "./client/dist",
}
`;

exports[`Test config maker test config object with all provided env variables 1`] = `
Object {
  "dbName": "test",
  "dbUri": "mongodb://localhost",
  "renderMode": "ssr",
  "serverPort": 1234,
  "spaPath": "../../../client/dist/_index.html",
  "ssrPath": "../../../client/dist/main.js",
  "staticRoot": "root",
}
`;

exports[`Test config maker test config object with only env.STATIC_ROOT variable 1`] = `
Object {
  "dbName": "hcard",
  "dbUri": "mongodb://192.168.99.101:27017/domain",
  "renderMode": "ssr",
  "serverPort": 3030,
  "spaPath": "../../../client/dist/_index.html",
  "ssrPath": "../../../client/dist/main.js",
  "staticRoot": "root",
}
`;
